mlflow/server/js/src/model-registry/components/SchemaTable.tsx

```diff
@@ -54,16 +54,44 @@ export class SchemaTableImpl extends React.PureComponent<Props> {
     );
   };
 
-  getSchemaTypeRepr = (schemaTypeSpec: any) => {
-    let { type } = schemaTypeSpec;
-    if (schemaTypeSpec.type === MODEL_SCHEMA_TENSOR_TYPE) {
-      type = `Tensor (dtype: ${schemaTypeSpec['tensor-spec'].dtype}, shape: [${schemaTypeSpec['tensor-spec'].shape}])`;
+  getSchemaTypeRepr = (schemaTypeSpec: ColumnSpec | TensorSpec): string => {
+    const { type } = schemaTypeSpec;
+
+    let repr: string = type;
+    if (schemaTypeSpec.type === 'tensor') {
+      repr = `Tensor (dtype: ${schemaTypeSpec['tensor-spec'].dtype}, shape: [${schemaTypeSpec['tensor-spec'].shape}])`;
+    } else {
+      repr = this.getColumnTypeRepr(schemaTypeSpec);
     }
 
     // If the "optional" property is present and true, wrap the type around an "Optional[]"
+    // NOTE: newer model signatures will have "required" instead of "optional", but we currently
+    // have to support both since both exist.
     if (schemaTypeSpec.optional) {
-      type = `Optional[${type}]`;
+      repr = `Optional[${repr}]`;
+    } else if (schemaTypeSpec.required) {
+      repr = `${repr} (required)`;
     }
+
+    return repr;
+  };
+
+  getColumnTypeRepr = (columnType: ColumnType): string => {
+    const { type } = columnType;
+
+    if (type === 'object') {
+      const propertyReprs = Object.keys(columnType.properties).map((propertyName) => {
+        const property = columnType.properties[propertyName];
+        const required = property.required ? ' (required)' : '';
+        return `"${propertyName}": ${this.getColumnTypeRepr(property)}${required}`;
```

Comment:
```suggestion
        return `${propertyName}: ${this.getColumnTypeRepr(property)}${required}`;
```

Can we remove `"`? I think it should look okay without it.

Reply:
We can! My only concern is that it might make it harder to distinguish properties from types if they have similar names, but that should be an edge case.